main/src/main/scala/sbt/internal/Load.scala

```diff
@@ -1012,21 +1014,21 @@ final case class LoadBuildConfiguration(
     extraBuilds: Seq[URI],
     log: Logger
 ) {
-  lazy val (globalPluginClasspath, _) = Load.pluginDefinitionLoader(this, Load.globalPluginClasspath(globalPlugin))
-
-  private[sbt] lazy val globalPluginDefs = {
-    val pluginData = globalPlugin match {
-      case Some(x) => PluginData(x.data.fullClasspath, x.data.internalClasspath, Some(x.data.resolvers), Some(x.data.updateReport), Nil)
-      case None    => PluginData(globalPluginClasspath, Nil, None, None, Nil)
+  lazy val globalPluginClasspath: Def.Classpath =
+    Load.buildPluginClasspath(this, Load.globalPluginClasspath(globalPlugin))
+
+  lazy val detectedGlobalPlugins: DetectedPlugins = {
+    val pluginData: PluginData = {
+      globalPlugin.map { info =>
+        val data = info.data
+        PluginData(data.fullClasspath, data.internalClasspath,
+          Some(data.resolvers), Some(data.updateReport), Nil)
+      }.getOrElse(PluginData(globalPluginClasspath))
```

Comment:
I generally write out pattern matching for `Option`. It's the fastest ("Not only is it by far the fastest (probably at least 10 times as fast as the alternatives), it is also the clearest." according to [Odersky](https://www.reddit.com/r/scala/comments/449td4/fold_or_pattern_matching_with_an_option/))